module/documents/actor/actor.mjs

```diff
@@ -3341,7 +3341,7 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
     const fill = CONFIG.DND5E.tokenHPColors[key];
 
     for ( const token of tokens ) {
-      if ( !token.object?.visible || !token.object?.renderable ) continue;
+      if ( !token.object?.visible || token.isSecret ) continue;
```

Comment:
`TokenDocument#isSecret` will be an addition with V12 Stable 4. Considering it is `undefined` in prior versions, we don't really need a version check here, I think.